File: src/modifiers/truncationMode.ts
import { createModifier } from './createModifier';

/// SwiftUI `Text.TruncationMode` values — where the ellipsis appears when
/// text is truncated.
export type TruncationMode = 'head' | 'middle' | 'tail';

export interface TruncationModeParams {
  mode?: TruncationMode;
}

/// SwiftUI `.truncationMode(_:)`. Controls where text within the view is
/// truncated when it doesn't fit. Pass a mode string directly or an object.
export function truncationMode(params?: TruncationModeParams | TruncationMode) {
  if (typeof params === 'string') {
    return createModifier('truncationMode', { mode: params });
  }
  return createModifier('truncationMode', params);
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/TruncationMode.swift
import SwiftUI

/// SwiftUI `.truncationMode(_:)`. Controls where text within the view is
/// truncated when it doesn't fit. Defaults to `.tail` (SwiftUI's default)
/// for an absent/unrecognized mode.
enum RNWTruncationModeModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("truncationMode") { view, params, _ in
            let mode: Text.TruncationMode
            switch params.string("mode") {
            case "head":   mode = .head
            case "middle": mode = .middle
            default:       mode = .tail
            }
            return AnyView(view.truncationMode(mode))
        }
    }
}
